stdlib/distutils/filelist.pyi

```diff
@@ -1 +1,80 @@
-class FileList: ...
+from typing import Callable, Iterable, List, Optional, Pattern, Union, overload
+from typing_extensions import Literal
+
+# class is entirely undocumented
+class FileList:
+    allfiles: Optional[Iterable[str]] = ...
+    files: List[str] = ...
+    def __init__(
+        self, warn: Optional[Callable[[str], None]] = ..., debug_print: Optional[Callable[[str], None]] = ...
+    ) -> None: ...
+    def set_allfiles(self, allfiles: Iterable[str]) -> None: ...
+    def findall(self, dir: str = ...) -> None: ...
+    def debug_print(self, msg: str) -> None: ...
+    def append(self, item: str) -> None: ...
+    def extend(self, items: Iterable[str]) -> None: ...
+    def sort(self) -> None: ...
+    def remove_duplicates(self) -> None: ...
+    def process_template_line(self, line: str) -> None: ...
+    @overload
+    def include_pattern(
+        self,
+        pattern: str,
+        anchor: Union[int, bool] = ...,
+        prefix: Optional[str] = ...,
+    ) -> bool: ...
+    @overload
+    def include_pattern(
+        self,
+        pattern: Union[str, Pattern[str]],
+        anchor: Union[int, bool] = ...,
+        prefix: Optional[str] = ...,
+        is_regex: Literal[0, False] = ...,
```

Comment:
I think this is backwards; we allow pattern to be a compiled regex if `is_regex` is True. Also if is_regex is True, the other arguments are ignored, so we shouldn't allow them to be passed.

Reply:
Fixed that one. I also added the is_regex True branch with positional args. Is that what you had in mind?

Reply:
Yup, that's what I had in mind :-) One more nit: I think the first overload of include_pattern / exclude_pattern is now redundant